Doc/c-api/module.rst

```diff
@@ -449,10 +449,11 @@ state:
    Add an object to *module* as *name*.  This is a convenience function which
    can be used from the module's initialization function.
 
-   On success, return ``0``. On error, raise an exception and return ``-1``.
+   On success, return ``0``.
 
-   Return ``NULL`` if *value* is ``NULL``. It must be called with an exception
-   raised in this case.
+   If *value* is ``NULL`` (indicating failure), raise an exception and return
+   ``-1``.  If an exception is set, raise that exception.  If no exception is
+   set, raise :exc:`SystemError`.
```

Comment:
This documentation sounds misleading to me. The normal case when value is NULL is that an exception is already set, so the function does **not** "raise an exception". Also, I dislike documenting the exact exception type for implementation details: please don't document "SystemError".

Reply:
> please don't document "SystemError".

That was my fault; I encouraged it in https://github.com/python/cpython/pull/101957#discussion_r1111313045. Sorry 'bout that!